icu4c/source/tools/icuexportdata/icuexportdata.cpp

```diff
@@ -293,3 +245,143 @@ int main(int argc, char* argv[]) {
 
     return 0;
 }
+
+#define INCLUDED_FROM_UCASE_CPP
+#include "ucase_props_data.h"
+
+struct AddRangeHelper {
+    UMutableCPTrie* ucptrie;
+};
+
+static UBool U_CALLCONV
+addRangeToUCPTrie(const void* context, UChar32 start, UChar32 end, uint32_t value) {
+    IcuToolErrorCode status("addRangeToUCPTrie");
+    UMutableCPTrie* ucptrie = ((const AddRangeHelper*) context)->ucptrie;
+    umutablecptrie_setRange(ucptrie, start, end, value, status);
+    handleError(status, "setRange");
+
+    return TRUE;
+}
+
+int exportCase() {
+    if (options[OPT_INDEX].doesOccur) {
+	fprintf(stderr, "--index option not supported for ucase\n");
+	return U_ILLEGAL_ARGUMENT_ERROR;
+    }
+    if (options[OPT_ALL].doesOccur) {
+	fprintf(stderr, "--all option not supported for ucase\n");
+	return U_ILLEGAL_ARGUMENT_ERROR;
+    }
+
+    IcuToolErrorCode status("icuexportdata");
+    LocalUMutableCPTriePointer builder(umutablecptrie_open(0, 0, status));
+    handleError(status, "exportCase");
+
+    const UTrie2* caseTrie = &ucase_props_singleton.trie;
+
+    AddRangeHelper helper = { builder.getAlias() };
+    utrie2_enum(caseTrie, NULL, addRangeToUCPTrie, &helper);
+
+    UCPTrieValueWidth width = UCPTRIE_VALUE_BITS_16;
+    LocalUCPTriePointer utrie(umutablecptrie_buildImmutable(
+        builder.getAlias(),
+        trieType,
+        width,
+        status));
+    handleError(status, "exportCase");
+
+    FILE* f = prepareOutputFile("ucase");
+
+    UVersionInfo versionInfo;
+    u_getUnicodeVersion(versionInfo);
+    char uvbuf[U_MAX_VERSION_STRING_LENGTH];
+    u_versionToString(versionInfo, uvbuf);
+    fprintf(f, "icu_version = \"%s\"\nunicode_version = \"%s\"\n\n",
+            U_ICU_VERSION,
+            uvbuf);
+
+    fputs("[[ucase]]\n\n", f);
```

Comment:
For the property data, I printed out array headers `[[ ]]` with the idea that the toml files could be concatenated together and then read in as a list with no conflicts.  Assuming there is only one single data file for collation, you probably shouldn't do the array thing.  (Just delete this line)

Reply:
Yeah, I ran into this when I started parsing the data and made the change locally.